src/test/scala/edu/berkeley/ce/rockslicing/RockSlicerSpec.scala

```diff
@@ -205,157 +204,212 @@ class RockSlicerSpec extends FunSuite {
+                                                             Seq.empty[Block])
+   val centroidMergedBlocks = mergedBlocks.map{ block =>
+     val centroid = block.centroid
+     Block(centroid, block.updateFaces(centroid))
+   }
+
+   assert(orphanBlocks.isEmpty)
+   assert(centroidMergedBlocks.length == 1)
+   assert(centroidMergedBlocks.head.approximateEquals(unitCube))
+ }
+
+ test("Three blocks should be merged into unit cube") {
+   val processorBlocks = Seq(leftQuarterUnitCube, centerPartUnitCube, rightQuarterUnitCube)
+   val (mergedBlocks, orphanBlocks) = RockSlicer.mergeBlocks(processorBlocks, Seq.empty[Block],
+                                                             Seq.empty[Block])
+   val centroidMergedBlocks = mergedBlocks.map{ block =>
+     val centroid = block.centroid
+     Block(centroid, block.updateFaces(centroid))
+   }
+
+   assert(orphanBlocks.isEmpty)
+   assert(centroidMergedBlocks.length == 1)
+   assert(centroidMergedBlocks.head.approximateEquals(unitCube))
+ }
+
+ test("Four blocks should be merged into unit cube") {
+   val processorBlocks = Seq(leftQuarterUnitCube, leftCenterQuarterUnitCube,
+                             rightCenterQuarterUnitCube, rightQuarterUnitCube)
+   val (mergedBlocks, orphanBlocks) = RockSlicer.mergeBlocks(processorBlocks, Seq.empty[Block],
+                                                             Seq.empty[Block])
+   val centroidMergedBlocks = mergedBlocks.map{ block =>
+     val centroid = block.centroid
+     Block(centroid, block.updateFaces(centroid))
+   }
+
+   assert(orphanBlocks.isEmpty)
+   assert(centroidMergedBlocks.length == 1)
+   assert(centroidMergedBlocks.head.approximateEquals(unitCube))
+ }
+
+ test("Eight blocks should be merged into two blocks that comprise top and bottom halves "+
+      "of unit cube") {
+   val processorBlocks = Seq(bottomLeftEighthUnitCube, bottomLeftCenterEighthUnitCube,
+                             bottomRightCenterEighthUnitCube, bottomRightEighthUnitCube,
+                             topLeftEighthUnitCube, topLeftCenterEighthUnitCube,
+                             topRightCenterEighthUnitCube, topRightEighthUnitCube)
+   val (mergedBlocks, orphanBlocks) = RockSlicer.mergeBlocks(processorBlocks, Seq.empty[Block],
+                                                             Seq.empty[Block])
+   val centroidMergedBlocks = mergedBlocks.map{ block =>
+     val centroid = block.centroid
+     Block(centroid, block.updateFaces(centroid))
+   }
+
+   val expectedBlocks = Seq(topHalfUnitCube, bottomHalfUnitCube)
+   val expectedBlocksCentroid = expectedBlocks map { block =>
+     val centroid = block.centroid
+     Block(centroid, block.updateFaces(centroid))
+   }
+
+   val blockCheck = centroidMergedBlocks.zip(expectedBlocksCentroid) map { case (calc, expected) =>
+     calc.approximateEquals(expected)
+   }
+   assert(orphanBlocks.isEmpty)
+   assert(centroidMergedBlocks.length == 2)
+   assert(!blockCheck.contains(false))
```

Comment:
You should just get rid of this and use an `assert` with a `forAll` on the line where you originally created `blockCheck`. So we can remove this `val` entirely.